Broken PR. Will be deleted

anton-bobkov opened the pull request:
This PR is BROKEN. It will be deleted

Commit:
Added the Troubleshooting section

blinkov commented on ydb/docs/en/core/dev/troubleshooting/schemas/overloaded-shards.md:
```diff
@@ -0,0 +1,23 @@
+
+* A table is created without the [AUTO_PARTITIONING_BY_LOAD](../../../concepts/datamodel/table.md#AUTO_PARTITIONING_BY_LOAD) clause.
+
+    In this case {{ ydb-short-name }} does not split overloaded shards. If a shard has to process too many queries, it will use up all its CPU resources and, consequently, query response times will increase.
```
It's not necessarily about CPU resources; it's about the fact that data shards are single-threaded and might not keep up with whatever is requested from them. Also, I'd describe this more carefully as there are a bit different cases: as far as I understand, `OVERLOADED` errors are returned more or less instantly when the load thresholds are exceeded (but you'll still lose time on retries), while the latency would be higher for queries that got accepted and wait for their turn to be executed (but they won't get the `OVERLOADED` error).

blinkov commented on ydb/docs/en/core/dev/troubleshooting/schemas/overloaded-shards.md:
```diff
@@ -0,0 +1,23 @@
+
+    In this case {{ ydb-short-name }} does not split overloaded shards. If a shard has to process too many queries, it will use up all its CPU resources and, consequently, query response times will increase.
+
+* A table is created with the [AUTO_PARTITIONING_MAX_PARTITIONS_COUNT](../../../concepts/datamodel/table.md#AUTO_PARTITIONING_MAX_PARTITIONS_COUNT) clause and has already reached the limit.
```
```suggestion
* A table was created with the [AUTO_PARTITIONING_MAX_PARTITIONS_COUNT](../../../concepts/datamodel/table.md#AUTO_PARTITIONING_MAX_PARTITIONS_COUNT) setting and has already reached the partition limit.
```

blinkov replied:
In SQL standard, queries consist of *statements* separated by semicolons (like `SELECT` statement, `INSERT INTO` statement, etc.), and *clauses* are syntactic parts of statements (`GROUP BY` clause, `ORDER BY` clause, etc.). So, in this case, there would be a `WITH` clause, and these `AUTO_PARTITIONING_MAX_PARTITIONS_COUNT` are more like settings for it.

blinkov commented on ydb/docs/en/core/dev/troubleshooting/schemas/overloaded-shards.md:
```diff
@@ -0,0 +1,23 @@
+
+{% include notitle [#](_includes/overloaded-shards-diagnostics.md) %}
+
+## Recommendations
+
+Consider the following solutions to the shard overload issue:
+
+* If the problem table is not partitioned by load, enable partitioning by load.
+
+* If the table has reached the maximum number of partitions, increase the limit.
```
```suggestion
## Recommendations

Consider the following solutions to address the shard overload issue:

* If the problem table is not partitioned by load, enable partitioning by load.

* If the table has reached the maximum number of partitions, increase the partition limit.
```

blinkov commented on ydb/docs/en/core/dev/troubleshooting/schemas/overloaded-shards.md:
```diff
@@ -0,0 +1,23 @@
+
+## Description
+
+[Data shards](../../../concepts/glossary.md#data-shard) in {{ ydb-short-name }} might get overloaded for the following reasons:
```
This article discusses several related issues; it probably makes sense to adjust the common article structure for this case.

Currently:

```text
# Header

## Description
...

- Issue 1
- Issue 2

## Diagnostics

...

## Recommendations

- Issue 1
- Issue 2
```

Could be:

```text
# Header

Introduction...

- [Issue 1](#issue-1)
- [Issue 2](#issue-2)

## Issue 1 {#issue-1}

More details on the issue.

### Diagnostics

How to check whether it applies or not.

### Recommendations

...

## Issue 2 {#issue-2}

More details on the issue.

### Diagnostics

How to check whether it applies or not.

### Recommendations

...

```

anton-bobkov replied:
We decided on a slightly different structure here with different recommendations for each case

blinkov commented on ydb/docs/en/core/dev/troubleshooting/troubleshooting-performance-issues.md:
```diff
@@ -0,0 +1,16 @@
+
+Addressing database performance issues often requires a holistic approach, including optimizing queries, configuring hardware resources correctly, and ensuring that both the database and application are well-designed. Regular monitoring and maintenance are key to identifying and resolving these issues proactively.
+
+## Tools to troubleshoot performance issues
+
+- {{ ydb-short-name }} metrics
+
+- {{ ydb-short-name }} logs
+
+- {{ ydb-short-name }} CLI
```
Add links

blinkov commented on ydb/docs/en/core/dev/troubleshooting/troubleshooting-performance-issues.md:
```diff
@@ -0,0 +1,16 @@
+
+## Overview
+
+Addressing database performance issues often requires a holistic approach, including optimizing queries, configuring hardware resources correctly, and ensuring that both the database and application are well-designed. Regular monitoring and maintenance are key to identifying and resolving these issues proactively.
```
```suggestion
Addressing database performance issues often requires a holistic approach, which includes optimizing queries, properly configuring hardware resources, and ensuring that both the database and the application are well-designed. Regular monitoring and maintenance are essential for proactively identifying and resolving these issues.
```

blinkov commented on ydb/docs/en/core/dev/troubleshooting/performance-issues.md:
```diff
@@ -0,0 +1,12 @@
+
+- **Query performance issues**. These issues refer to problems that arise when database queries execute slower that expected because of their inefficient design.
+
+
```
Another potential major category is YDB operating in a degraded mode, like during a major outage or drills.

Commit:
Apply suggestions from code review

Co-authored-by: Ivan Blinkov <ivan@blinkov.ru>

Commit:
Restructured the topic and added classification of performance issues

blinkov commented on ydb/docs/en/core/dev/troubleshooting/performance/queries/transaction-lock-invalidation.md:
```diff
@@ -0,0 +1,20 @@
+
+{% include notitle [#](_includes/transaction-lock-invalidation.md) %}
+
+## Recommendations
+
+The longer a transaction lasts, the higher the likelihood of encountering a *transaction locks invalidated* error.
+
+Avoid interactive transactions. For example, try to avoid the following pattern:
+
+1. Select some data.
+1. Process the selected data in the application.
+1. Update some data in the database.
+1. Commit the transaction in a separate query.
+
+A better approach is to use a single YQL query to select data, update data, and commit the transaction.
```
```suggestion
## Recommendations

The longer a transaction lasts, the higher the likelihood of encountering a *transaction locks invalidated* error.

If possible, avoid interactive transactions. For example, try to avoid the following pattern:

1. Select some data.
1. Process the selected data in the application.
1. Update some data in the database.
1. Commit the transaction in a separate query.

A better approach is to use a single YQL query to select data, update data, and commit the transaction.
```

blinkov commented on ydb/docs/en/core/dev/troubleshooting/performance/queries/uncached-queries.md:
```diff
@@ -0,0 +1,7 @@
+# Uncached queries
+
+
+
+## Diagnostics
+
+## Recommendation
```
Empty

blinkov commented on ydb/docs/en/core/dev/troubleshooting/performance/schemas/_includes/overloaded-shards-diagnostics.md:
```diff
@@ -0,0 +1,41 @@
+# Diagnostics of overloaded shards
+
+1. Analyze the **Overloaded shard count** chart in the **DB overview** Grafana dashboard.
+
+    ![](../_assets/overloaded-shards-dashboard.png)
+
+    The chart shows whether the {{ ydb-short-name }} cluster has overloaded shards, but it does not indicate which specific table shards are overloaded.
+
+1. To identify the table with an overloaded shard, follow these steps:
+
+    1. In the [Embedded UI](../../../../../reference/embedded-ui/index.md), go to the **Databases** tab and click the database.
+
+    1. On the **Navigation** tab, ensure the entire database is selected.
+
+    1. Open the **Diagnostics** tab.
+
+    1. Open the **Top shards** tab.
+
+    1. In the **Immediate** and **Historical** tabs, sort the shards by the **CPUCores** column and analyze the information.
```
```suggestion
# Diagnostics of overloaded shards

1. Analyze the **Overloaded shard count** chart in the **DB overview** Grafana dashboard.

    ![](../_assets/overloaded-shards-dashboard.png)

    The chart indicates whether the {{ ydb-short-name }} cluster has overloaded shards, but it does not specify which table shards are overloaded.

1. To identify the table with an overloaded shard, follow these steps:

    1. In the [Embedded UI](../../../../../reference/embedded-ui/index.md), go to the **Databases** tab and click on the database.

    1. On the **Navigation** tab, ensure the entire database is selected.

    1. Open the **Diagnostics** tab.

    1. Open the **Top shards** tab.

    1. In the **Immediate** and **Historical** tabs, sort the shards by the **CPUCores** column and analyze the information.
```

blinkov commented on ydb/docs/en/core/dev/troubleshooting/performance/schemas/_includes/overloaded-shards-diagnostics.md:
```diff
@@ -0,0 +1,41 @@
+
+        If the table does not have these options, see [Recommendations for table configuration](../overloaded-shards.md#table-config)
+
+1. Analyze whether primary key values autoincrement monotonically.
```
How?

blinkov commented on ydb/docs/en/core/dev/troubleshooting/performance/schemas/_includes/overloaded-shards-diagnostics.md:
````diff
@@ -0,0 +1,41 @@
+
+    ![](../_assets/partitions-by-cpu.png)
+
+1. To pinpoint the schema issue, follow these steps:
+
+    1. Retrieve information about the problematic table using the [{{ ydb-short-name }} CLI](../../../../../reference/ydb-cli/index.md). Run the following command:
+
+        ```bash
+        ydb scheme describe <table_name>
+        ```
+
+    1. In the command output, analyze the **Auto partitioning settings**:
+
+        * `Partitioning by size`
+        * `Partitioning by load`
+        * `Max partitions count`
+
+        If the table does not have these options, see [Recommendations for table configuration](../overloaded-shards.md#table-config)
+
+1. Analyze whether primary key values autoincrement monotonically.
+
+    If they do, see [Recommendations for the imbalanced primary key](../overloaded-shards.md#pk-recommendations)
````
```suggestion
1. To pinpoint the schema issue, follow these steps:

    1. Retrieve information about the problematic table using the [{{ ydb-short-name }} CLI](../../../../../reference/ydb-cli/index.md). Run the following command:

        ```bash
        ydb scheme describe <table_name>
        ```

    2. In the command output, analyze the **Auto partitioning settings**:

        * `Partitioning by size`
        * `Partitioning by load`
        * `Max partitions count`

        If the table does not have these options, see [Recommendations for table configuration](../overloaded-shards.md#table-config).

2. Analyze whether primary key values increment monotonically.

    If they do, see [Recommendations for the imbalanced primary key](../overloaded-shards.md#pk-recommendations).
```

blinkov commented on ydb/docs/en/core/dev/troubleshooting/performance/schemas/overloaded-shards.md:
```diff
@@ -0,0 +1,29 @@
+# Overloaded shards
+
+## Description
```
I thought we've removed such headers from other articles? Let's make it consistent.

blinkov commented on ydb/docs/en/core/dev/troubleshooting/performance/schemas/overloaded-shards.md:
```diff
@@ -0,0 +1,29 @@
+
+## Description
+
+[Data shards](../../../../concepts/glossary.md#data-shard) in {{ ydb-short-name }} might get overloaded for the following reasons:
+
+* A table is created without the [AUTO_PARTITIONING_BY_LOAD](../../../../concepts/datamodel/table.md#AUTO_PARTITIONING_BY_LOAD) clause.
+
+    In this case {{ ydb-short-name }} does not split overloaded shards. If a shard has to process too many queries, it will use up all its CPU resources and, consequently, query response times will increase.
+
+* A table was created with the [AUTO_PARTITIONING_MAX_PARTITIONS_COUNT](../../../../concepts/datamodel/table.md#AUTO_PARTITIONING_MAX_PARTITIONS_COUNT) setting and has already reached the partition limit.
+
+* Inefficient primary key that results in the imbalance of queries between shards. A typical example is ingestion with a monotonically increasing (autoincrementing) primary key that might overload the "last" partition.
```
```suggestion
[Data shards](../../../../concepts/glossary.md#data-shard) in {{ ydb-short-name }} may become overloaded for the following reasons:

* A table is created without the [AUTO_PARTITIONING_BY_LOAD](../../../../concepts/datamodel/table.md#AUTO_PARTITIONING_BY_LOAD) clause.

    In this case, {{ ydb-short-name }} does not split overloaded shards. If a shard has to process too many queries, it will consume all its CPU resources, leading to increased query response times.

* A table was created with the [AUTO_PARTITIONING_MAX_PARTITIONS_COUNT](../../../../concepts/datamodel/table.md#AUTO_PARTITIONING_MAX_PARTITIONS_COUNT) setting and has already reached its partition limit.

* An inefficient primary key that causes an imbalance in the distribution of queries across shards. A typical example is ingestion with a monotonically increasing primary key, which may overload the "last" partition. For example, this could occur with an autoincrementing primary key using the [serial](../../../../yql/reference/types/serial.md) data type.
```

blinkov commented on ydb/docs/en/core/dev/troubleshooting/performance/schemas/overloaded-shards.md:
```diff
@@ -0,0 +1,29 @@
+
+{% include notitle [#](_includes/overloaded-shards-diagnostics.md) %}
+
+## Recommendations for table configuration {#table-config}
+
+Consider the following solutions to address the shard overload issue:
+
+* If the problem table is not partitioned by load, enable partitioning by load.
+
+* If the table has reached the maximum number of partitions, increase the partition limit.
+
+## Recommendations for the imbalanced primary key {#pk-recommendations}
+
+Consider changing the primary key to evenly distribute the load between table partitions.
```
```suggestion
## Recommendations for table configuration {#table-config}

Consider the following solutions to address shard overload:

* If the problematic table is not partitioned by load, enable partitioning by load.

* If the table has reached the maximum number of partitions, increase the partition limit.

## Recommendations for the imbalanced primary key {#pk-recommendations}

Consider modifying the primary key to distribute the load evenly across table partitions.
```

blinkov replied:
We've discussed that the primary keys aren't that easy to change

Commit:
Apply suggestions from code review

Co-authored-by: Ivan Blinkov <ivan@blinkov.ru>

Commit:
WIP

Commit:
Apply suggestions from code review

Commit:
Added an article on system clock drift

Commit:
WIP

vladl2802 committed:
Make block combine hashed use stream instead of flow (#9979)

Enjection committed:
Treat backup collections as dirs (#10490)

maximyurchuk committed:
ya.make fixes (missing RECURSE) (#10454)

maximyurchuk committed:
Mute ydb/core/mind/hive/ut/THiveTest.DrainWithHiveRestart (#10514)

naspirato committed:
Fix in test report mute string (#10515)

MBkkt committed:
Vector index coordination [2/N] (#10462)

maximyurchuk committed:
Fix sed in fail summary (exclude brackets in content) (#10520)

iddqdex committed:
increase timeout on tpch1000 q09 (#10492)

pashandor789 committed:
[CBO] Improve Transitive Closure + Cycle processing added (#10121)

abyss7 committed:
Mitigate double notification of compiled pattern scenario (#10499)

Commit:
FBP-None: nots.py add MORE typings

Побочный рефакторинг, давно напрашивался, плюс помог мне чуть освоится в этом файле.
commit_hash:c8dbe9573b9d41f85ad255ba9470cb93bf87d69c

VestniK committed:
Remove unused TEST_CLASSPATH_ORIGINS test-dart key
commit_hash:b50bd183d64278fd25272a91d7f68db07e598d78

CyberROFL committed:
ImportData: consistent limits (#10524)

Commit:
Reload local config after update
commit_hash:f1c246c8ab871f8c7c36aaef08f2c18ad63540d2

georgthegreat committed:
Remove more unused sources from contrib/tools/m4
commit_hash:eed3e8af9d590cc0006acb8b733df8027b97e3ce

georgthegreat committed:
contrib/tools/m4: Copy contrib/tools/bison solution to get rid of isnan{d,f,l}.c sources
commit_hash:31788279593d5a121af9f7d229957edabf1773e9

Commit:
YT-23193: Add wrapper for error of read meta timeout
commit_hash:a7df9bc8696f8e9dfc8acb7e8695046ebec2b47c

Commit:
[build] Add optional SUFFIX to BUNDLE()
commit_hash:340460c18f39c151147e8e942a5e82e8f485f714

Commit:
Fix MinRowCountPerSubquery options getting lost
commit_hash:9260aa4e245870829fc2040cae423ab4c6891360

va-kuznecov committed:
Use new SafeEntropyPoolRead which fills entire buffer (#10534)

robot-piglet committed:
Update contrib/libs/lzma to 5.6.3
commit_hash:625353c0ed27ce033147a8f5121982b83e126bdc

georgthegreat committed:
Revert more patches applied to contrib/tools/m4
commit_hash:fb87ade017a656215b6a71fc4504fb2c07c3b370

Enjection committed:
Allow system columns in tables inside backup collections (#10523)

Commit:
New version of the tld SKIP_CHECK SKIP_REVIEW
commit_hash:a96b155c1ea432bcff49800f63149d1ff83f071e

authorius committed:
More mod ordering
commit_hash:a6f9325e26045597decde41f1fe1a18707c1224f

Commit:
remove unused includes in util/generic/string.h
commit_hash:edd5ed00db3a4c6090c535fa406f184268188733

zaverden committed:
feat(conf+builder): remove contrib-related code
commit_hash:cfbd74bf877107b5db0cb2ea60977b4849e15af5

nepal committed:
Stabilize table order in test (#10539)

robot-piglet committed:
Intermediate changes
commit_hash:2d9a1d3a3501b70a635f24b69a195c83059b71da

robot-piglet committed:
Update contrib/libs/c-ares to 1.34.2
commit_hash:abd65f143c484666e1545e6bbfb9d947df95ecd6

adameat committed:
do not convert google datetime to string in schema (#10487)

APozdniakov committed:
[YQ-3761] Fix RewriteAsHoppingWindow optimization (#10455)

Commit:
Fix disabling ref counted tracking
commit_hash:8552424a0dfce80a71c406632beba95646bf5b3e

mxkovalev committed:
YQL: Improve logs in DQ worker (#10504)

ItIsApachee committed:
Add default options to advance queue consumer
commit_hash:b7a4848fd251739854804af7bd3db3956a60c25e

evanevanevanevannnn committed:
Additional validation for queries with "json_list" format and datetime types (#10208)

dcherednik committed:
[refactoring] Get rid of grpc global request limiter copy-paste code. (#10518)

kruall committed:
Add workflow for publishing dstool (#9787)

gridnevvvit committed:
remove kicli tests (#10537)

ijon committed:
base: move TCompactionPolicy out of flat_scheme_op.proto (#10495)

Move `TCompactionPolicy` from `flat_scheme_op.proto` to distinct .proto file.

The final goal is to break dependency of `ydb/core/base/` on `flat_scheme_op.pb.h`.
Currently, `flat_scheme_op.proto` contains few fundamental structures essential for `ydb/core/base/`, alongside numerous higher-level components (scheme entities descriptions, scheme operations) that are irrelevant to `ydb/core/base/`'s needs.

Gazizonoki committed:
Increment YDB CLI version to 2.15.0 (#10546)

MBkkt committed:
Vector index coordination [3/N] (#10498)

mxkovalev committed:
YQL: Add DependsOn for ToFlow callable (#10457)

maximyurchuk committed:
Correct mime type for err/out in ya make tests results (#10385)

kruall committed:
Fix workflow for uploading dstool (#10550)

nikitka committed:
ci: add an "external" label for issues that were opened by external users (#10551)

yumkam committed:
streamlookup: pass and use MaxDelayedRows parameter (#10489)

pixcc committed:
Add backup/restore test for all scheme objects (#10448)

azevaykin committed:
Datashard follower statistics (#10482)

Gazizonoki committed:
Added links to bins in downloads for CLI 2.11, 2.12, 2.13, 2.14, 2.15 (#10549)

pashandor789 committed:
[CBO] Transitive closure adding conditions to existed edges (#10528)

Commit:
YT-23134: Fix sensor service paths
commit_hash:6c52404e8c00e60be2ab7545af30a4232874971a

maximyurchuk committed:
Script for showing unreachable targets (#10453)

va-kuznecov committed:
Set SubmitInFlightBytesMax back to 32KiB (#10565)

Commit:
YTORM-1187: Fix IN and BETWEEN expression work with indexes
commit_hash:12d86de033f73ea238ebde7f47b689d30014119a

yumkam committed:
Streamlookup special case nulls in keys (#10508)

dcherednik committed:
[refactoring] Do not include executor_thread.h into actor.h (#10562)

dcherednik committed:
Public api to return node id in the describe table call (#10571)

Motivation:
Network is limited, we need to reduce amount of transfers between nodes.

One of problem:
Session actor has no knowledge about key range distribution (data shard locations). Which means we have to transfer data between data shard and kqp executor to perform read or write.

Workaround:
Provide estimation about keyRange -> nodeId distribution to use session (which means executor and other kqp components) created on the same node where located corresponding data shard.

vladl2802 committed:
Revert "Make block combine hashed use stream instead of flow (#9979)" (#10575)

mregrock committed:
Revert "Removed legacy KV API" (#10552)

pashandor789 committed:
[CBO] No join was chosen bug fix (#10576)

Co-authored-by: Pavel Ivanov <pudge1000-7@qavm-9f0570a4.qemu>

azevaykin committed:
Follower stats reuses some params from leader: DataSize, RowCount, IndexSize (#10561)

yumkam committed:
Add streamlookup monitoring (dynamic counters) (#10280)

aavdonkin committed:
Fixed clearing table stats after alter operaion (#10509)

koct9i committed:
Couple fixes for clangd language server

- yt/client: include async_stream.h in file_reader.h
  NApi::IFileReader needs NConcurrency::IAsyncZeroCopyInputStream.

- Declare yt/yt/client/api/rpc_proxy/protocol_version_variables.h using CONFIGURE_FILE
  Otherwise resulting file does not materialize after:
  ya make --build-all --force-build-depends --replace-result --add-result=.h yt/yt/client/

  Proper source file generation is required at least for making clangd happy.

---

Pull Request resolved: https://github.com/ytsaurus/ytsaurus/pull/894
commit_hash:b27f7963bc113543c0ac3842f78636be4f33416f

ssmike committed:
Prefer indices aligned with order-by-limit (#10584)

alexvru committed:
Fix replication bug (#10560)

kardymonds committed:
YQ-3767 Shared reading: fix field names with @ (#10563)

naspirato committed:
Computing days in row for test analytics (#10163)

kardymonds committed:
YQ-3766 Shared reading: clear used_size by stop_session (#10586)

CyberROFL committed:
Add include system/event.h (#10581)

adameat committed:
add node name filter, improve filter behavior (#10585)

nshestakov committed:
Fix autopartitioning of topics with path that is not root of dat… (#10593)

alephonea committed:
CBO cost function for YT (#9333)

Co-authored-by: Dmitriy Orlov <orlovorlov@dnorlov-dev.vla.yp-c.yandex.net>

rvu1024 committed:
[yt provider] Use intermediate streams (#10598)

kruall committed:
Change step to common step for publishing pachage (#10602)

ijon committed:
build_bloat: add paths sorting to improve visual stability (#10604)

Sorting paths enables the renderer to produce more consistent visual results across different runs, even when build statistics change. Elements with minimal duration change between runs are more likely to maintain their positions on a grid.

This stability makes it easier to visually compare the outcomes of build time optimization efforts aimed at specific modules or components.

kungasc committed:
Improve Memory Controller data size logs (#10574)

kruall committed:
Change dstool's version to 0.0.12 (#10608)

MBkkt committed:
Adjust vector index settings (#10542)

pashandor789 committed:
[CBO] TEST ASAN FIX (#10610)

pixcc committed:
Enhance help for ImportData option in ydb tools restore (#10599)

abyss7 committed:
Return export of dynamic symbols (#10617)

maximyurchuk committed:
mute ydb/library/yql/providers/generic/connector/tests/datasource/clickhouse test.py.test_select_datetime[datetime_string_NATIVE-*] (#10622)

va-kuznecov committed:
Revert noop scheduler related commits (#10600)

shnikd committed:
Get next val by path id in SequencerActor (#10519)

jepett0 committed:
Enable subqueries inside views (#10517)

jepett0 committed:
fix import to arcadia (#10590)

apollo1321 committed:
Fix issue

Fix issue
commit_hash:b0190c0b56d0be237ab4b54a94574adc607f0965

dorooleg committed:
compute id has been supported (#9556)

vitstn committed:
Fixed IN with tuple input in some cases (#10605)

maximyurchuk committed:
Mute ydb/core/kqp/workload_service/ut ResourcePoolClassifiersDdl.TestCreateResourcePoolClassifierOnServerless (#10629)

maximyurchuk committed:
Merge pull request #10502 from ydb-platform/mergelibs-241016-1210

Library import 241016-1210

nshestakov committed:
More test for autopartitionin of topics (#10613)

Commit:
add generic json command executor for external bs
commit_hash:0833419ebee283a1fb0e0a41f615b0bf7b61f548

adameat committed:
correctly pass, merge and check body parameters in local-rpc calls (#10601)

sourcecd committed:
add some queue for nodes operations (#10620)

Co-authored-by: Sergey J <wint@yandex-team.ru>

georgthegreat committed:
contrib/tools/m4: Remove windows-specific libc replacements
commit_hash:ec4d153d072b7dc65470431459aa432df6b2e82a

Commit:
WIP

Commit:
WIP

robot-piglet committed:
Intermediate changes
commit_hash:1338f91346a59e6bbff8c8f2444e990a51dc2e63

Commit:
drop api/rpc_proxy/protocol_version_variables.h.in from SRCS
commit_hash:ed0b5a509549b7f0ebf81240aa688d15d670c8e1

Commit:
fix for py3 runner
commit_hash:30adefe665fb83ed7b1aa644958e110348e0d117

alexv-smirnov committed:
Merge branch 'rightlib' into mergelibs-241019-1758

alexv-smirnov committed:
Import libraries 241019-1758

georgthegreat committed:
Put contrib/tools/m4 under yamaker
commit_hash:0fad455f3d7eb62f50aece2733dbdc32b2484c62

maximyurchuk committed:
Merge pull request #10642 from ydb-platform/mergelibs-241019-1758

Library import 241019-1758

gridnevvvit committed:
remove deprecated & unused result lib (#10641)

robot-piglet committed:
Intermediate changes
commit_hash:eddaace1b99679850506a2fe1993b3e0be396422

Commit:
New version of the tld SKIP_CHECK SKIP_REVIEW
commit_hash:514e533e0d7a227fbca23493746cfdcfa224756b

robot-piglet committed:
Intermediate changes
commit_hash:15bc9967e5232ba2f5c2b66acdb48018622376b4

georgthegreat committed:
Update contrib/tools/m4 to 1.4.18
commit_hash:dab43d0051ce5656201a6f031d378c04dceefa2c

Nechda committed:
Add suffix for each platform
commit_hash:0722662b4c536ce15948e436775b09615a49592f

Commit:
Introduce message_level_overrides to tune log message levels at runtime

* Changelog entry
Type: feature

Add message_level_overrides option to logging config for better run-time tuning.
commit_hash:07e9563fd111c437edf7ac0e5dd190781878d8fa

alexander-somov committed:
Introduce CUDA_DEVICE_LINK_LIBRARY
commit_hash:7b188b7472202555da5b6607b32737377294e2a6

robot-piglet committed:
Intermediate changes
commit_hash:42cbd3d6db0c8793544ceb2bba9ad42631021f52

Commit:
Automatic release build for ymake, os_ymake

From hash: [62803d3d6d0d464354ad85728c92d544bb92b621](HIDDEN_URL
From revision: [15128769](HIDDEN_URL
[CI flow](HIDDEN_URL
Flow triggered by user: [robot-ci](HIDDEN_URL

Update tools: ymake, os_ymake
commit_hash:097e72efe2f2e7302d07f936c8715fe43ba811de

robot-piglet committed:
Intermediate changes
commit_hash:04a153065be0050f5e911b9ff8b6dccbfe1b3cc8

nemo-cpt committed:
Dismantle owner and subscriber processing
commit_hash:1b53252a36a6ef28835bdfa289dfe0746fcd61b7

ulya-sidorina committed:
fix(doc): add info about objects types that allowed for export (#10627)

Co-authored-by: Ivan Blinkov <ivan@ydb.tech>

san-kir-k committed:
Tuple layout pack/unpack (#4905)

Co-authored-by: Vyacheslav Boev <slava.boev94@gmail.com>

GrigoriyPA committed:
YQ RD fix import contirb build (#10606)

jepett0 committed:
Index build: do not lose the requested partitioning info of indexImplTables in case of SchemeShard reboots (#10579)

adameat committed:
fix slots per disk not returned without a filter (#10607)

georgthegreat committed:
Fix building contrib/tools/{bison,m4} with musl-libc

Also: put devstack_targets.inc under musl autocheck.
commit_hash:a824666c706272699d21127ad3d73694357f07ae

iddqdex committed:
Add tests for tpc {h | ds} data generator (#10564)

Commit:
Automatic release build for test_tool, ya_bin, os_ya, os_test_tool

From hash: [1efd96d04601c45d4d379ff72f1fc4883960a11d](HIDDEN_URL
From revision: [15133324](HIDDEN_URL
[CI flow](HIDDEN_URL
Flow triggered by user: [robot-ci](HIDDEN_URL

Update tools: test_tool, ya_bin, os_ya, os_test_tool
commit_hash:45ab2b7d3082ec5d68b5dd41d8b9baa9d2f286e5

Enjection committed:
Align create backup collection entries grammar with corresponding topics grammar (#10580)

Commit:
Rewrite THiveManager to offload all transient activities to background thread
commit_hash:5a8d9467975843f579713f79f3f00f88c836a50b

alexv-smirnov committed:
Merge branch 'rightlib' into mergelibs-241021-1020

alexv-smirnov committed:
Import libraries 241021-1020

Commit:
Update github.com/ydb-platform/fq-connector-go to 0.5.11-rc.5 (#10654)

kardymonds committed:
YQ-3743 Add message to Y_ABORT_UNLESS in checkpoint_coordinator (#10660)

uzhastik committed:
fix FindWrapStats for wide lambda (#10662)

Hor911 committed:
Handle and log all stat processing issues uniformly (#10376)

maximyurchuk committed:
Merge pull request #10656 from ydb-platform/mergelibs-241021-1020

Library import 241021-1020

gridnevvvit committed:
TEvQueryResponse supports arenas in simplier manner (#9771)

pashandor789 committed:
[CBO] Remove shared_ptr<> from Statistics (#10625)

ijon committed:
schemeshard: split schemeshard__operation_common.cpp, move code from .h (#10631)

Move operations specific content of schemeshard__operation_common.cpp (bsv, cdc_stream, pq, subdomain) into separate files.
Move dependency rich methods definitions out of schemeshard__operation_common.h.

This is part of "improve schemeshard operations build time" effort (#10633).

swalrus1 committed:
Use simdjson for binary json construction for improved performance (#10464)

vitstn committed:
Less deps in yql/utils (#10667)

Тест флаки, с асаном общая проблема

iddqdex committed:
Fix statistics path when use cli (#10683)

blinkov committed:
[docs] manual deployment refactoring (#10078)

Co-authored-by: Andrey Fomichev <fomichev@ydb.tech>

blinkov commented on ydb/docs/en/core/dev/troubleshooting/performance/system/system-clock-drift.md:
```diff
@@ -0,0 +1,46 @@
+# System clock drift
+
+Synchronized clocks are critical for distributed databases. If system clocks on the {{ ydb-short-name }} servers start to drift too much, distributed transactions might be processed with higher latencies. In some cases {{ ydb-short-name }} might even fail to process distributed transactions and return errors.
+
+If a {{ ydb-short-name }} cluster contains multiple [coordinators](../../../../concepts/glossary.md#coordinator), planned transactions are merged by [mediators](../../../../concepts/glossary.md#mediator).
+
+If the system clocks of the nodes that run coordinator tablets differ, transaction latencies grow by the time difference between the fastest and the slowest system clocks. The latency grows because a transaction planned on a node, which system clock is faster, will be executed only after the coordinator with the slowest wall clock reaches that time.
+
+Moreover, if the system clock drift exceeds 30 seconds, {{ ydb-short-name }} might fail to process distributed transactions. Before coordinators start planning a transaction, affected DataShards determine an acceptable range of timestamps for the transaction. The start of the time range is the current clock time of the mediator tablet. The end time is determined by the planning timeout of 30 seco ...
```
```suggestion
# System clock drift

Synchronized clocks are critical for distributed databases. If system clocks on the {{ ydb-short-name }} servers drift excessively, distributed transactions will experience increased latencies.

If a {{ ydb-short-name }} cluster contains multiple [coordinators](../../../../concepts/glossary.md#coordinator), planned transactions are merged by [mediators](../../../../concepts/glossary.md#mediator) before being sent off for execution.

If the system clocks of the nodes running the coordinator tablets differ, transaction latencies increase by the time difference between the fastest and slowest system clocks. This occurs because a transaction planned on a node with a faster system clock can only be executed once the coordinator with the slowest clock reaches the same time.

Furthermore, if the system clock drift exceeds 30 seconds, {{ ydb-short-name }} will refuse to process distributed transactions. Before coordinators start planning a transaction, affected DataShards determine an acceptable range of timestamps for the transaction. The start of this range is the current time of the mediator tablet's clock, while the 30-second planning timeout determines the end. If the coordinator's system clock exceeds this time range, it cannot plan a distributed transaction for the DataShard, resulting in errors for such queries.
```

blinkov commented on ydb/docs/en/core/dev/troubleshooting/performance/system/system-clock-drift.md:
```diff
@@ -0,0 +1,46 @@
+
+    The system clock drift problems will be listed under `NODES_TIME_DIFFERENCE`.
+
+{% note info %}
```
I'd not put these into a note as these are all equal places to look at, and it will likely take comparing a few (if not all of them) to figure out what's actually going on.

blinkov commented on ydb/docs/en/core/dev/troubleshooting/performance/system/system-clock-drift.md:
```diff
@@ -0,0 +1,46 @@
+
+1. On the **Info** tab, click the **Healthcheck** button.
+
+    If the **Healthcheck** button has the `MAINTENANCE REQUIRED` status, the {{ ydb-short-name }} cluster might have some problems, including the system clock drift. The diagnosed problems will be listed in the **DATABASE** section under the **Healthcheck** button.
```
It feels reasonable to add a link to https://ydb.tech/docs/en/reference/ydb-sdk/health-check-api, which covers this in much more detail.

blinkov replied:
```suggestion
    If the **Healthcheck** button displays a `MAINTENANCE REQUIRED` status, the {{ ydb-short-name }} cluster might have problems, such as system clock drift. Any identified issues will be listed in the **DATABASE** section beneath the **Healthcheck** button.
```
"Problem" and "issue" aren't always interchangeable. Issues are more generic, while problems are only severe issues that need attention.

blinkov commented on ydb/docs/en/core/dev/troubleshooting/performance/system/system-clock-drift.md:
```diff
@@ -0,0 +1,46 @@
+
+- Open the [Interconnect overview](../../../../reference/embedded-ui/interconnect-overview.md) page of the [Embedded UI](../../../../reference/embedded-ui/index.md).
+
+- Use such tools as `pssh` or `ansible` to run the command on all {{ ydb-short-name }} nodes to display the system clock value.
```
This needs extra clarification:

- Let's highlight that network delays between the host that runs pssh/ansible and YDB hosts will influence the results.
- Let's recommend some specific command to get precise timestamps like `date +%s%N`.
- As we've discussed on the previous iteration, these daemons for time synchronization you mention below on the last line of this file have diagnostics tools that can tell how well do they think they're doing their job, for example, `timedatectl show-timesync --all`. That might be easier to reason about compared to raw timestamps.

blinkov commented on ydb/docs/en/core/dev/troubleshooting/performance/system/system-clock-drift.md:
```diff
@@ -0,0 +1,46 @@
+
+1. Sync the system clocks of {{ ydb-short-name }} nodes manually. For example, you can use `pssh` or `ansible` to run the clock sync command on all of the nodes.
+
+1. Ensure that system clocks on all of the {{ ydb-short-name }} servers are synced by `ntpd` or `chrony`. It's recommended to use the same time source for all servers in the {{ ydb-short-name }} cluster.
```
The one that comes with https://ydb.tech/docs/en/devops/ansible/initial-deployment is `timesyncd`, let's add it to the list.

blinkov commented on ydb/docs/en/core/dev/troubleshooting/performance/system/tablets-moved.md:
```diff
@@ -0,0 +1,76 @@
+
+{{ ydb-short-name }} automatically balances the load by transferring tablets from overloaded nodes to other nodes. This process is managed by [Hive](*hive). When Hive moves tablets, queries to those tablets suffer higher latencies.
+
+<!-- This information is taken from a draft topic Concepts > Hive. -->
```
Where's this draft? Let's review it there rather than do it two times.

blinkov commented on ydb/docs/en/core/dev/troubleshooting/performance/system/tablets-moved.md:
```diff
@@ -0,0 +1,76 @@
+# Frequent tablet transfers between nodes
+
+{{ ydb-short-name }} automatically balances the load by transferring tablets from overloaded nodes to other nodes. This process is managed by [Hive](*hive). When Hive moves tablets, queries to those tablets suffer higher latencies.
```
```suggestion
{{ ydb-short-name }} automatically balances the load by transferring tablets from overloaded nodes to other nodes. This process is managed by [Hive](*hive). When Hive moves tablets, queries affecting those tablets might experience increased latencies while they wait for the tablet to get initialized on the new node.
```

blinkov commented on ydb/docs/en/core/dev/troubleshooting/performance/system/tablets-moved.md:
```diff
@@ -0,0 +1,76 @@
+
+
+
+[*counter]: Virtual resource that is used for balanncing tablets, which do not have other hardware resource metrics (CPU, memory, network), and for tablets of column-based tables. If a tablet has this resource, its value always equals to `1`.
+
+[*hive]: A Hive is a system tablet responsible for launching and managing other tablets. Its responsibilities include moving tablets between nodes in case of node failure or overload.
```
Let's just link to the glossary instead of creating local definitions.

anton-bobkov replied:
Linked Hive to a glossary. However, counter is not described in the glossary. So I kept the local definition for now.

anton-bobkov replied:
Linked Hive to a glossary. However, counter is not described in the glossary. So I kept the local definition for now.

blinkov replied:
I'd rather update the glossary right in this PR.

blinkov commented on ydb/docs/en/core/dev/troubleshooting/performance/system/tablets-moved.md:
```diff
@@ -0,0 +1,76 @@
+
+1. To reduce the chances of balancing, increase the following Hive balancer thresholds:
+
+    - MinCounterScatterToBalance
+    - MinCPUScatterToBalance
+    - MinMemoryScatterToBalance
+    - MinNetworkScatterToBalance
+    - MaxNodeUsageToKick
+    - ObjectImbalanceToBalance
```
This better be a table with some description, defaults, allowed and practical ranges, etc.

blinkov commented on ydb/docs/en/core/dev/troubleshooting/performance/system/ydb-updates.md:
```diff
@@ -0,0 +1,3 @@
+# {{ ydb-short-name }} updates
+
+{{ ydb-short-name }} is a distributed system that supports [rolling restart](../../../../devops/manual/upgrade.md), when database administrators update {{ ydb-short-name }} nodes one by one. This helps keep the {{ ydb-short-name }} cluster up and running during the update process. However, when a {{ ydb-short-name }} node is being restarted, Hive moves the tables that run on this node to other no ...
```
It sounds like a copy-paste of some text above, and the details are missing.

ssmike committed:
Force precompute for returning effect input (#10231)

Commit:
Apply suggestions from code review

Co-authored-by: Ivan Blinkov <ivan@blinkov.ru>

Commit:
Apply suggestions from code review

Co-authored-by: Ivan Blinkov <ivan@blinkov.ru>

snaury committed:
Fix volatile transactions getting stuck after a restart (#10679)

alexv-smirnov committed:
Do not include python codegen into cmake codegen (#10697)

alexv-smirnov committed:
Pregenerate ydb/core/base/generated for cmake build (#10699)

CyberROFL committed:
CDC initial scan settings (#10536)

anton-bobkov commented on ydb/docs/en/core/dev/troubleshooting/index.md:
```diff
@@ -0,0 +1 @@
+# Troubleshooting
```
It contains a single link to performance/index.md for now

Commit:
Enable duplicating logs to stderr (#10595)

Co-authored-by: Dmitry Razumov <dvrazumov@yandex-team.ru>

maximyurchuk committed:
Remove asan sqs from PR (#10702)

naspirato committed:
Mute ydb/core/blobstorage/ 4 tests (#10687)

iddqdex committed:
Canonical results for tpc-h (#10669)

adameat committed:
improve whiteboard tracing and skip invalid path id (#10704)

niksaveliev committed:
Fix kafka cdc read and login without @ (#10678)

lll-phill-lll committed:
Wide combiner spill during state split (#9785)

azevaykin committed:
Statistics: disable ANALYZE in KQP without feature flag (#10706)

snaury committed:
Implement runtime flags codegen in C++ (#10711)

azevaykin committed:
Partition at decimal keys (#10696)

alexv-smirnov committed:
Bump clang version to 16 for cmake build (#10715)

Commit:
New field COMPRESSION_LEVEL in Column Family (#10645)

shmel1k committed:
[log_backend] write tenant & cluster for UnifiedAgent log backend (#10666)

robdrynkin committed:
Remove print from dstool cluster balance (#10733)

APozdniakov committed:
[YQ-3621] support AFTER MATCH SKIP PAST LAST ROW (#10597)

alexvru committed:
Unmute Osiris tests and temporarily disable multithreading (#10737)

vitstn committed:
Don't use Now() during replay for yt settings (#10732)

mxkovalev committed:
YQL: Skip Hybrid for false UseSystemColumns pragma (#10710)

adameat committed:
add MemoryDetailed field to viewer/nodes handler (#10694)

UgnineSirdis committed:
Support pure json output for audit log (#10143)

kardymonds committed:
YQ-3763  Fix read count (try to fix flaky test) (#10644)

anton-bobkov commented on ydb/docs/en/core/dev/troubleshooting/performance/system/_includes/tablets-moved.md:
```diff
@@ -0,0 +1,33 @@
+# Diagnosing frequent tablet transfers
```
Let's use moves (simpler)

snaury committed:
Add counters for volatile transactions (#10723)

niksaveliev committed:
Allow short auto partitioning strategies yql (#10701)

anton-bobkov commented on ydb/docs/en/core/dev/troubleshooting/queries/overloaded-errors.md:
```diff
@@ -0,0 +1,19 @@
+
+## Diagnostics
+
+{% include notitle [#](_includes/overloaded-errors.md) %}
```
Added a step to see the number of sessions and a step that refers to overloaded shards

CyberROFL committed:
Introduce logging in 'tools restore' (#10681)

alexv-smirnov committed:
Bump tooling clang version to 16 for cmake build (#10756)

Commit:
ydb_topic: Call non-dataEvent callbacks immediately (#10751)

gridnevvvit committed:
query service: optimize result passing over trailing results (#10690)

ElenaAfina committed:
autopartitioning-eng (#10532)

Co-authored-by: Ivan Blinkov <ivan@ydb.tech>

Alek5andr-Kotov committed:
Default value for option `--codec` (#10651)

FloatingCrowbar committed:
Stabilize DR tests (#10750)

azevaykin committed:
Decimal and PG support in viewer (#10752)

uzhastik committed:
increase grpc call timeout in TMonitoringGrpcServiceActor (#10734)

nshestakov committed:
Fix wrong active and inactive partition number (#10712)

alexv-smirnov committed:
Fix typo in clang.toolchain (#10769)

kardymonds committed:
YQ-3766 Shared reading: add unread stats (#10635)

Enjection committed:
Add coop create cdc evaluation to copy/create table calls (#10503)

Commit:
AlterColumnTable (#10672)

yumkam committed:
streamlookup: fix zero default for MaxDelayedRows (#10735)

abyss7 committed:
Improve logging and errors verbosity (#10768)

vladl2802 committed:
Fixed: Make block combine use stream instead of flow (#10707)

jepett0 committed:
Fix a bug in YQL node creation for indexImplTable table settings (#10741)

CyberROFL committed:
Get rid of unused code, cleanup dependencies (#10777)

Commit:
persqueue: add user-agent counters (#10603)

jepett0 committed:
Add decimal split boundaries backup / restore test (#10790)

nshestakov committed:
Fix balancing of partitions for special reading sessions (#10779)

kungasc committed:
Do not block all requests on Shared Cache policy switch (#10619)

rvu1024 committed:
Disable cbo_4tables test (#10798)

vitstn committed:
Recover from an exception inside building of dq yson result (#10795)

iddqdex committed:
Fix iterations reporting (#10792)

naspirato committed:
Unmute stable tests (#10760)

GrigoriyPA committed:
YQ-3784 support computation graph invalidation in purecalc (#10799)

ivanmorozov333 committed:
compaction speedup (#10323)

Commit:
YDB FQ: avoid using `clickhouse-connect` library in integration tests (#10776)

alexv-smirnov committed:
Specify only ydb/apps to be built by cmake postcommit check (#10808)

anton-bobkov commented on ydb/docs/en/core/dev/troubleshooting/performance/hardware/_includes/cpu-bottleneck.md:
```diff
@@ -0,0 +1,13 @@
+
+1. Compare the following Grafana charts:
+
+    - **DB overview** > **Latency** > **Read only tx server latency**
+
+        ![](../_assets/cpu-read-only-tx-latency.png)
```
This topic is rewritten from scratch

alexv-smirnov committed:
Temporary patch to dix difference between antlr4.9 and 4.13 behaviour in cmake build (#10816)

kardymonds committed:
YQ-3775 Shared reading: try to fix memory leak in topic session (#10786)

UgnineSirdis committed:
Fix memory consumption restriction in stream RPC calls (#10758)

nikitka committed:
ci: add clang-16, antlr 4.13.2 to Github Runner VM Image (#10806)

GrigoriyPA committed:
YQ WM fixed flaky tests (#10805)

kardymonds committed:
YQ-3779 Remove FlagSubscribeOnSession (use TEvSubscribe) (#10720)

lex007in committed:
Add escaping for serialized pg types in viewer (#10778)

ivanmorozov333 committed:
fix tx volume calculation for inserted chunks (#10813)

ivanmorozov333 committed:
chunks count for limit in compaction (#10812)

ivanmorozov333 committed:
correct limit for indexation (count of chunks) (#10811)

ivanmorozov333 committed:
fix tx writing limit (#10810)

alexv-smirnov committed:
Append TOKEN antlr patch to non-sanitizer build (#10829)

ivanmorozov333 committed:
fix error processing on program apply (#10814)

dorooleg committed:
PoolId has been renamed to ResourcePool in c++ sdk (#10770)

alexv-smirnov committed:
Fix ccache credentials for cmake build (#10833)

alexv-smirnov committed:
Return C compiler env var to cmake build job (#10834)

nshestakov committed:
Support for autopartitioning topics for CLI (#10832)

naspirato committed:
Unmute stable tests (#10830)

aavdonkin committed:
Added counters for local db tables loading times (#10402)

loochek committed:
Block input for YT map operations (#10587)

ivanmorozov333 committed:
prefetch necessary tables before loading (#10809)

alexv-smirnov committed:
build: refresh Embedded UI (v6.28.0) (#10836)

Co-authored-by: artemmufazalov <artemmufazalov@users.noreply.github.com>

alexv-smirnov committed:
Fix username link for ccache (#10847)

maximyurchuk committed:
Remove autoconfig asan tests from PR (#10843)

iddqdex committed:
fix windows build (#10841)

maximyurchuk committed:
Remove fq/plans asan tests from PR (#10845)

iddqdex committed:
fix darwin build (#10851)

Enjection committed:
Extract prefix and entries in backup-related sql (#10807)

yumkam committed:
Fix input transform destructor (#7909)

nikvas0 committed:
Improve tx defer (#10507)

loochek committed:
Sys columns support for YT map block input mode (#10844)

CyberROFL committed:
Encapsulate BackupFolder() in DumpClient (#10804)

mregrock committed:
Legacy kv api remove (#10614)

Remove legacy KV API and LocalEnumerateTablets

dorooleg committed:
Compute logs improvement (#10856)

Commit:
Working on comments (part 1)

Commit:
Working on comments (part 1)

Commit:
Blurred domain

Commit:
Merge branch 'docs-troubleshooting-performance' of github.com:anton-bobkov/ydb into docs-troubleshooting-performance

anton-bobkov commented on ydb/docs/en/core/dev/troubleshooting/performance/schemas/_includes/splits-merges.md:
```diff
@@ -0,0 +1,49 @@
+
+1. To identify recently split or merged tables, follow these steps:
+
+    1. In the [Embedded UI](../../../../../reference/embedded-ui/index.md), go to the **Nodes** tab and select a node.
```
Moved the comment

anton-bobkov commented on ydb/docs/en/core/dev/troubleshooting/performance/schemas/_includes/overloaded-shards-diagnostics.md:
```diff
@@ -0,0 +1,49 @@
+
+        * `Partitioning by size`
+        * `Partitioning by load`
+        * `Max partitions count`
```
Moved the comment

anton-bobkov commented on ydb/docs/en/core/dev/troubleshooting/performance/infrastructure/hardware.md:
```diff
@@ -0,0 +1,26 @@
+
+- **Availability of nodes on racks**
+
+    On the **Nodes** tab, see if nodes on specific racks are not available. Analyze the health indicators in the **Host** and **Rack** columns.
```
Moved the comment

anton-bobkov commented on ydb/docs/en/core/dev/troubleshooting/performance/index.md:
```diff
@@ -0,0 +1,77 @@
+
+- **OS and YDB-related issues**.
+
+    - **[YDB updates](system/ydb-updates.md)**. YDB is a distributed system that supports rolling restart, when database administrators update YDB nodes one by one. This helps keep the YDB cluster up and running during the update process. However, when a YDB node is being restarted, Hive moves the tables that run on this node to other nodes, and that may lead to increased latencies for queries th ...
```
Moved the comment

anton-bobkov commented on ydb/docs/en/core/dev/troubleshooting/performance/index.md:
```diff
@@ -0,0 +1,69 @@
+
+    - **[YDB updates](../../../devops/manual/upgrade.md)**. YDB is a distributed system that supports rolling restart, when database administrators update YDB nodes one by one. This helps keep the YDB cluster up and running during the update process. However, when a YDB node is being restarted, Hive moves the tables that run on this node to other nodes, and that may lead to increased latencies fo ...
+
+    - **Hardware resource allocation issues**. Suboptimal allocation of resources, for example poorly configured control groups (cgroups), may result in insufficient resources for {{ ydb-short-name }} and increase query latencies even though physical hardware resources are still available on the database server.
```
Moved the comment

anton-bobkov commented on ydb/docs/en/core/dev/troubleshooting/performance/hardware/io-bandwidth.md:
```diff
@@ -0,0 +1,3 @@
+# I/O bandwidth
+
+High read/write operations can overwhelm disk subsystems, leading to increased latencies in data access. When the system cannot read or write data quickly enough, queries that require disk access will be delayed.
```
Added the article
